include/aspect/plugins.h

```diff
@@ -323,18 +325,16 @@ namespace aspect
         // verify that the same name has not previously been
         // used to register a plugin, since we would then no
         // longer be able to identify the plugin
-        for (typename std::list<PluginInfo>::const_iterator
-             p = plugins->begin();
-             p != plugins->end(); ++p)
-          Assert (std::get<0>(*p) != name,
+        for (const auto &p : *plugins)
+          Assert (std::get<0>(p) != name,
                   ExcMessage ("A plugin with name <" + name + "> has "
                               "already been registered!"));
 
         // now add one record to the list
-        plugins->push_back (PluginInfo(name,
-                                       description,
-                                       declare_parameters_function,
-                                       factory_function));
+        plugins->emplace_back (PluginInfo(name,
```

Comment:
```suggestion
        plugins->emplace_back (name,
```
Otherwise emplace_back will do the same as push_back.

Reply:
Yes, good eye! So fixed!